python_ta/contracts/__init__.py

```diff
@@ -414,6 +425,10 @@ def _check_invariants(instance, klass: type, global_scope: dict) -> None:
             _debug(f"Warning: could not evaluate representation invariant: {invariant}")
         else:
             if not check:
+                # Remove the "currenlty_checking" attribute before printing the error message to the user
```

Comment:
It's a bit cumbersome to duplicate the "delattr" code. You are trying to achieve the behaviour of "regardless of what happens inside this loop, make sure `delattr` is called".

The cleanest way to do this is by using an outer `try` block that has a `finally` clause (you can look up what `finally` does if you haven't used it before).